src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -451,6 +451,50 @@ private function isSimpleInstance($id, Definition $definition)
     }
 
     /**
+     * Checks if the definition is a trivial instance.
+     *
+     * @param Definition $definition
+     *
+     * @return bool
+     */
+    private function isTrivialInstance(Definition $definition)
+    {
+        if ($definition->getMethodCalls() || $definition->getProperties() || $definition->getConfigurator() || $definition->isLazy()) {
+            return false;
+        }
+        if ($definition->isDeprecated() || $definition->getFactory() || 3 < \count($definition->getArguments())) {
+            return false;
+        }
+
+        foreach ($definition->getArguments() as $arg) {
+            if (!$arg || ($arg instanceof Reference && 'service_container' !== (string) $arg)) {
```

Comment:
because this considers `new Foo($this)` a trivial case ($this being the dumped container reference of course)